### ISSUE TITLE
option to change roles visitor <-> participant

### ISSUE THREAD

**Issue by rms-joe** [iNPUTmice/Conversations#3888]
#### General information

* **Version:** 2.9.0
* **Device:** Sony XZ1c
* **Android Version:**  Android 9 (stock)
* **Server name:** disroot.org
* **Server software:** Prosody trunk nightly build 1337 (2020-09-17, f7f30a3464fe) 
* **Installed server modules:** https://compliance.conversations.im/server/disroot.org/
* **Conversations source:** F-Droid

#### Steps to reproduce

1. create public channel (A)
2. join channel with another account (B)
3. owner(A) change channel to moderated
4. B is now "no affiliation (participant)" in a moderated room
5. to change his role, A can set B to member and revoke afterwards.
6. B is now "no affiliation (visitor)

7. owner(A) change channel back to unmoderated
8. B is still "no affiliation (visitor)
9. to change his role, A can set B to member and revoke afterwards.
10. B is now "no affiliation (participant)"

#### Expected result

it would make sense to have an option to change roles here.
root cause is that "no affiliation" is bundled with different default roles in unmoderated and moderated channels.

in my opinion, a good solution would be to add "grant voice" and "revoke voice" to the manage privileges menu,
that changes role between "participant" and "visitor".

https://xmpp.org/extensions/xep-0045.html

cheers, avg_joe

**Comment by iNPUTmice** [comment]
If you make someone a member in a moderated role they get voice; but permanently, right? That seems to be enough? And you don’t have to redo this operation after they have been offline for a second.

**Comment by rms-joe** [comment]
If someone gets offline, he rejoins with default role.
What if I want to set my unmoderated channel to moderated temporarily, cause things are escalating... 
I see a chance here to introduce a basic moderation feature plus solving the not so elegant status quo.

**Comment by iNPUTmice** [comment]
The only difference between the affiliation member and the role participant is the permanency? You can already give someone permanent 'voice'. I don’t see a reason of introducing a temporary one on top of that. (Where the duration of that temporal state is not all transparent to the average user)

**Comment by rms-joe** [comment]
Yeah, I'm more heading toward to revoke someone's voice temporarily.
